server/datastore/mysql/vpp.go

```diff
@@ -498,22 +543,22 @@ WHERE vat.global_or_team_id = ? AND va.title_id = ?
 }
 
 func (ds *Datastore) InsertHostVPPSoftwareInstall(ctx context.Context, hostID uint, appID fleet.VPPAppID,
-	commandUUID, associatedEventID string, selfService bool,
+	commandUUID, associatedEventID string, selfService bool, policyID *uint,
 ) error {
 	stmt := `
 INSERT INTO host_vpp_software_installs
-  (host_id, adam_id, platform, command_uuid, user_id, associated_event_id, self_service)
+  (host_id, adam_id, platform, command_uuid, user_id, associated_event_id, self_service, policy_id)
 VALUES
-  (?,?,?,?,?,?,?)
+  (?,?,?,?,?,?,?,?)
 	`
 
 	var userID *uint
-	if ctxUser := authz.UserFromContext(ctx); ctxUser != nil {
+	if ctxUser := authz.UserFromContext(ctx); ctxUser != nil && policyID == nil {
```

Comment:
Wanna make sure I understand this `policyID == nil` check here.

Is it because if there is no policy ID present that means that this install was done by a person and
not by Fleet automatically?

Reply:
The converse of that. If we see a policy ID associated with an install, it can't have been done by a person, so make sure we never attribute the install to a person.